### PR TITLE
WIP feat: initial r/profile dapp

### PR DESCRIPTION
**Pull request by moul**
This PR intends to let users define custom metadata that can be reused across
the whole ecosystem.

This realm is intended to be used by people that owns a r/user entry or
not (works well with just an `std.Address`).

This is a preliminary realm for bigger projects that would to rely on Gno,
including a twitter clone, or maybe an ENS that could store IP addresses, etc.

The goal is to pack the contract with default fields that are the expected most common ones for customization (profile_picture, name, primary_color, is_dev (to enable more debug), etc. But the contract should support a way to set new fields, basically a key/value store associated with a profile.

Exected usage API from another contract:

```go
import "gno.land/r/demo/profile"

func MyHandler() {
    caller := std.GetOrigCaller()
    settings := profile.ByAddr(caller)
    username, _ := settings.Get(profile.FieldUsername)
    customField, _ := settings.Get("my-custom-key")
}
```

---

Depends on #145 
Depends on #149

### CODE REVIEW

**Comment by thehowl** [comment]
Are you still interested in working on this? Shall we keep it open? @moul

**Comment by moul** [comment]
Yes, I will continue.

**Comment by moul** [comment]
@n0izn0iz needs something like this for Teritori.

@gfanton you were asking for contract ideas for your onboarding, I think this one makes sense.

File: examples/gno.land/r/profile/profiles.gno [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+	"strings"
+
+	"gno.land/p/avl"
+	"gno.land/p/ufmt"
+	"gno.land/r/users"
+)
+
+// TODO: makes sense to implement an allowance system?
+
+var profiles *avl.MutTree // std.Address.String() -> *Profile
```
Comment (moul) [inline examples/gno.land/r/profile/profiles.gno:14]: using the old avl API

**Comment by moul** [comment]
Quick update:
I suggest making this realm function like Go context, allowing any package/realm to define a new type with a key and generate new entries. Additionally, I aim to have the storage cost controlled by the calling package/realm. I am still considering the best approach for this.

**Comment by moul** [comment]
ping @gnolang/24-zone01-1

**Pull request closed without merging**